### ISSUE TITLE
Update default statistics

### ISSUE THREAD

**Issue by anetasie** [sherpa/sherpa#226]
The current default statistics is set to chi2gehrels. It is a biased statistics defined for analysis of X-ray spectra and it is not really good for general fitting of the data.
Poisson likelihood should be default for X-ray analysis - we have a choice of cstat, wstat and cash. 
wstat has not been well tested, while cstat and cash require modeling of the background data.
I would vote for cstat until we have a better confidence in wstat performance.

For general analysis Gaussian data we should use chi2datavar. This allows for subtraction and would error out if the background subtracted data goes negative.

**Comment by DougBurke** [comment]
At present we do not have a concept of a default statistic that depends on the "type" of the data set (i.e. to switch between `chi2datavar` and `cstat`). I think that would be hard to add on any sort of reasonable time scale (as there's a lot of issues about how this would work at the `sherpa.*.ui` level: i.e. could the statistic change each time I call `ui.load_data`). 

It would actually be slightly easier to add to the low-level `Fit` object (but it would rely on being able to ask a Data object what it's "type" is, and we do not have a reliable way to determine this in the general case), but this might cause inconsistencies at the UI level, so I would be worried about doing this on the 4.9 release schedule.

It is very easy to change the default (for both the UI layer and the `Fit` object) from `chi2gehrels` to `chi2datavar`. That is, do not have a "default poisson" case.

**Comment by anetasie** [comment]
We should make a change from `chi2gehrels` to `chi2datavar` for 4.9.

**Comment by anetasie** [comment]
Given issues with ```chi2datavar``` I suggest that we set the default statistics in sherpa to ```cstat``` and the method to ```simplex``` for 4.9 release.

**Comment by hamogu** [comment]
The 4.9 release has passed and so did 4.10, 4.11, ... and we did not yet change the defaults. Should we? If not, we should close this issue instead...
Personally, I actually suggest to leave everything as is. My understanding is that `cstat` can be a problem if you don't have Poisson data, e.g. you some dataset that contains negative numbers. The current `chi2gehrels` will work even then.

So, I vote to close this issue and instead keep an eye out for writing docs the help users to select the right statistics.

**Comment by anetasie** [comment]
`chi2geherls` will work in the sense that the user does not see any failure of the code, but the resulting parameters are often biased, even for the data with high number of counts. Of course even the high number of counts in the X-ray observations are Poisson. Should we support non-Poisson data analysis as default while most of the CIAO users work with Poisson data?

**Comment by hamogu** [comment]
I don't think so. I think the current default is somewhat sensible in the sense that it works for a wide variety of data sets. Sure, it's biased in some cases, but there is no statistic that's not biased for some dataset. Switching to `cash` will lead to outright failures for any data that's negative, thus I believe that it's not a good default and I advocate to leave the default what it's now and to close this issue, which suggested "changing the default statistics".

The proper solution is probably not to have a default at all and let the user pick a statistic consciously every time. That's the case for the OO already. In the UI, that ship has probably sailed.
